packages/react-dom/src/events/__tests__/SyntheticKeyboardEvent-test.js

```diff
@@ -451,6 +451,30 @@ describe('SyntheticKeyboardEvent', () => {
         });
       });
     });
+
+    describe('code', () => {
+      it('should throw (in dev) when KeyboardEvent.code is undefined', () => {
+        const node = ReactDOM.render(
+          <input
+            onKeyDown={e => {
+              return e.code;
+            }}
+          />,
+          container,
+        );
+        let evt = new KeyboardEvent('keydown', {
+          code: undefined,
+          bubbles: true,
+          cancelable: true,
+        });
+
+        if (__DEV__ && evt.code === undefined) {
+          expect(() => node.dispatchEvent(evt)).toThrowError();
```

Comment:
This should use the infrastructure we have in place for dev only warnings. Check out [this](https://github.com/facebook/react/blob/679402a66b09e129d06f415cd976d18727cc8590/packages/react/src/__tests__/ReactJSXElementValidator-test.js#L229-L235) and [this](https://github.com/facebook/react/blob/679402a66b09e129d06f415cd976d18727cc8590/packages/react/src/ReactElementValidator.js#L300-L308).

This will also make sure that the warning is _not_ emitted in production and that the message is indeed correct.